libs/auth-nest-tools/src/lib/jwt.strategy.ts

```diff
@@ -6,20 +6,29 @@ import { passportJwtSecret } from 'jwks-rsa'
 import type { AuthConfig } from './auth.module'
 import { JwtPayload } from './jwt.payload'
 import { Auth } from './auth'
+import { multiIssuerKeyProvider } from './multi-issuer-key-provider'
 
 const AUTH_BODY_FIELD_NAME = '__accessToken'
 const JWKS_URI = '/.well-known/openid-configuration/jwks'
+const keyProviderBaseOptions = {
+  cache: true,
+  rateLimit: true,
+}
 
 @Injectable()
 export class JwtStrategy extends PassportStrategy(Strategy) {
   constructor(private config: AuthConfig) {
     super({
-      secretOrKeyProvider: passportJwtSecret({
-        cache: true,
-        rateLimit: true,
-        jwksUri: `${config.issuer}${JWKS_URI}`,
-      }),
-
+      secretOrKeyProvider: Array.isArray(config.issuer)
+        ? multiIssuerKeyProvider({
+            ...keyProviderBaseOptions,
+            jwksUri: JWKS_URI,
+            issuers: config.issuer,
+          })
+        : passportJwtSecret({
+            ...keyProviderBaseOptions,
+            jwksUri: `${config.issuer}${JWKS_URI}`,
+          }),
```

Comment:
Would be nice to move all of this to a function like `secretOrKeyProvider: createKeyProvider(config.issuer),` which has all of the options and jwks path logic.